pkg/scalers/github_runner_scaler.go

```diff
@@ -403,15 +416,58 @@ func parseGitHubRunnerMetadata(config *ScalerConfig) (*githubRunnerMetadata, err
 	}
 
 	if val, ok := config.AuthParams["personalAccessToken"]; ok && val != "" {
-		// Found the organizationURL in a parameter from TriggerAuthentication
-		meta.personalAccessToken = val
+		// Found the pat token in a parameter from TriggerAuthentication
+		meta.personalAccessToken = &val
+	}
+
+	if appID, instID, key, err := setupGitHubApp(meta, config); err == nil {
+		meta.applicationID = appID
+		meta.installationID = instID
+		meta.applicationKey = key
 	} else {
-		return nil, fmt.Errorf("no personalAccessToken given")
+		return nil, err
+	}
+
+	if meta.applicationKey == nil && meta.personalAccessToken == nil {
+		return nil, fmt.Errorf("no personalAccessToken or appKey given")
 	}
 
 	meta.scalerIndex = config.ScalerIndex
 
-	return &meta, nil
+	return meta, nil
+}
+
+func setupGitHubApp(meta *githubRunnerMetadata, config *ScalerConfig) (*int64, *int64, *string, error) {
+	var appID *int64
+	var instID *int64
+	var appKey *string
+
+	if val, err := getInt64ValueFromMetaOrEnv("applicationID", config); err == nil && val != -1 {
+		appID = &val
+	}
+
+	if val, err := getInt64ValueFromMetaOrEnv("installationID", config); err == nil && val != -1 {
+		instID = &val
+	}
+
+	if val, ok := config.AuthParams["appKey"]; ok && val != "" {
+		appKey = &val
+	}
+
+	if (appID != nil || instID != nil || appKey != nil) &&
+		(appID == nil || instID == nil || appKey == nil) {
+		return nil, nil, nil, fmt.Errorf("applicationID, installationID and applicationKey must be given")
+	}
+
+	if meta.applicationID != nil && meta.installationID != nil && meta.applicationKey != nil {
+		httpTrans := kedautil.CreateHTTPTransport(false)
+		_, err := gha.New(httpTrans, *meta.applicationID, *meta.installationID, []byte(*meta.applicationKey))
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("error creating GitHub App client: %w, \n appID: %d, instID: %d", err, appID, instID)
+		}
+	}
```

Comment:
Do we need to test this here? I mean, we are also checking it during the client creation

Reply:
let me run another test in the morning, I put it in specifically to fix something just can't remember what

Reply:
Yeah, that's a good question. It is not something critical, but would be nice to know if it is really necessary.

Reply:
I've removed the pre check. I was in there to fix a bug, which I subsequently fixed elsewhere and didn't remove it